### PR TITLE
[OSD-17712] Add more logging for Account Creation Failure

### PR DESCRIPTION
**Pull request by T0MASD**
# What is being added?
On CreateAccount error, gets createAccount failure reasons and uses LogAwsError helper to output formatted log

## Checklist before requesting review

- [x] I have tested this locally
- [x] I have included unit tests
- [ ] I have updated any corresponding documentation

## Steps To Manually Test
_Please provide us steps to reproduce the scenarios needed to test this.  If an integration test is provided, let us know how to run it. If not, enumerate the steps to validate this does what it's supposed to._
1. run make test

Ref [OSD-17712](https://issues.redhat.com//browse/OSD-17712)

### CODE REVIEW

**Comment by bergmannf** [comment]
I guess we have to merge https://github.com/openshift/aws-account-operator/pull/776 first and rebase afterwards.
Seems the only complained about unclean working directory is the different image tag.

File: api/v1alpha1/account_types.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ type AccountStatus struct {
 	Claimed       bool   `json:"claimed,omitempty"`
 	SupportCaseID string `json:"supportCaseID,omitempty"`
 	// +optional
+        // +listType=map
```
[the remark this answers is not in the document]
Comment (T0MASD) [reply api/v1alpha1/account_types.go:87]: this was added in cb6903ab63f1ca930f2df0260904a2306cc09f8c
to address :

```                                                                                                                                                                                
API rule violation: list_type_missing,github.com/openshift/aws-account-operator/api/v1alpha1,AccountStatus,Conditions 
```
which was firing during `./boilerplate/_lib/container-make generate-check`  when executing

```
openapi-gen --logtostderr=true -i ./api/v1alpha1 -o  -O zz_generated.openapi -p ./api/v1alpha1 -h /dev/null -r -   
```

**Pull request merged**